variants/NUCLEO_F756ZG/variant.cpp

```diff
@@ -0,0 +1,218 @@
+/*
+  Copyright (c) 2011 Arduino.  All right reserved.
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#include "pins_arduino.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Pin number
+const PinName digitalPin[] = {
+  PG_9,  //D0
+  PG_14, //D1
+  PF_15, //D2
+  PE_13, //D3
+  PF_14, //D4
+  PE_11, //D5
+  PE_9,  //D6
+  PF_13, //D7
+  PF_12, //D8
+  PD_15, //D9
+  PD_14, //D10
+  PA_7,  //D11
+  PA_6,  //D12
+  PA_5,  //D13
+  PB_9,  //D14
+  PB_8,  //D15
+  PC_6,  //D16
+  PB_15, //D17
+  PB_13, //D18
+  PB_12, //D19
+  PA_15, //D20
+  PC_7,  //D21
+  PB_5,  //D22
+  PB_3,  //D23
+  PA_4,  //D24
+  PB_4,  //D25
+  PB_6,  //D26
+  PB_2,  //D27
+  PD_13, //D28
+  PD_12, //D29
+  PD_11, //D30
+  PE_2,  //D31
+  PA_0,  //D32
+  PB_0,  //D33 - LED_GREEN
+  PE_0,  //D34
+  PB_11, //D35
+  PB_10, //D36
+  PE_15, //D37
+  PE_14, //D38
+  PE_12, //D39
+  PE_10, //D40
+  PE_7,  //D41
+  PE_8,  //D42
+  PC_8,  //D43
+  PC_9,  //D44
+  PC_10, //D45
+  PC_11, //D46
+  PC_12, //D47
+  PD_2,  //D48
+  PG_2,  //D49
+  PG_3,  //D50
+  PD_7,  //D51
+  PD_6,  //D52
+  PD_5,  //D53
+  PD_4,  //D54
+  PD_3,  //D55
+  PE_2,  //D56
+  PE_4,  //D57
+  PE_5,  //D58
+  PE_6,  //D59
+  PE_3,  //D60
+  PF_8,  //D61
+  PF_7,  //D62
+  PF_9,  //D63
+  PG_1,  //D64
+  PG_0,  //D65
+  PD_1,  //D66
+  PD_0,  //D67
+  PF_0,  //D68
+  PF_1,  //D69
+  PF_2,  //D70
+  PA_7,  //D71
+  NC,    //D72
+  PB_7,  //D73 - LED_BLUE
+  PB_14, //D74 - LED_RED
+  PC_13, //D75 - USER_BTN
+  PD_9,  //D76 - Serial Rx
+  PD_8,  //D77 - Serial Tx
+  PA_3,  //D78/A0
+  PC_0,  //D79/A1
+  PC_3,  //D80/A2
+  PF_3,  //D81/A3
+  PF_5,  //D82/A4
+  PF_10, //D83/A5
+  PB_1,  //D84/A6
+  PC_2,  //D85/A7
+  PF_4,  //D86/A8
+  PF_6   //D87/A9
+};
+
+// Analog (Ax) pin number array
+const uint32_t analogInputPin[] = {
+  78, //A0
+  79, //A1
+  80, //A2
+  81, //A3
+  82, //A4
+  83, //A5
+  84, //A6
+  85, //A7
+  86, //A8
+  87, //A9
+  11, //A10
```

Comment:
analogInputPin[] to be updated after taking into account variant.h comment.

Reply:
Not even sure what I'm supposed to do.

Reply:
In variant.h I commented that PD14 is not an analog pin so it could not be named A10 (A means Analog in this define) . In the same way   `11, //A10` is not valid as A10 is not valid.

Reply:
Can't make heads or tails looking at the data sheets and even comparing against other datasheets and trying to reverse engineer in a way. Am I looking at the wrong documents? Or what should I be looking for that determines this? https://www.st.com/en/microcontrollers-microprocessors/stm32f756zg.html

Reply:
This as nothing to do with datasheet, it is linked to Arduino way of working.
All pins with ADC and DAC capabilities should be declared as Analog Ax. Thus it correspond to all pins listed in `PinMap_ADC[]` and `PinMap_DAC[]`. PD14 is not listed in those arrays.